ADR/0034-workspace-experience.md

```diff
@@ -0,0 +1,41 @@
+# 34. Workspace Experience
+
+## Status
+
+WiP
+
+## Context
+
+We inherited the "workspace" concept from our early days when we expected to be integrated with the [KCP] project. Long ago (over a year at the time of this writing), we decided to remove [KCP] from the architecture of the Konflux project - but, we have never revisited the workspace and defined and decided what we want it to be and what we want it to mean.
+
+Workspaces are an important concept in the Konflux user experience today. The user onboards into a workspace of their own. A team member of theirs may add them to a team workspace. As a platform administrator, if you hear about a user having trouble, you'll ask them what workspace they're working in.
+
+Workspace roles give a way to describe who has what rights in the workspace.
+
+Workspaces happen to be our mechanism for managing quota - a fact inherited from the [kubesaw] stack.
+
+Workspaces provide a kind of transparent multi-cluster for us. A user's workspace may be allocated to one cluster or another - but our proxy layer keeps the user blissfully unaware of this. Workspaces can be placed on clusters either automatically (using [kubesaw] mechanisms) or manually by a platform administrator.
+
+Workspaces also provide a kind of general user-management today. Workspaces and users are mixed up - we can't tell which workspaces correspond with real users and which workspaces are team-based workspaces. They all look like users. However, they _do_ give us a concrete representation of a user, which we can use to power the invite dialog to add other users to your workspace.
+
```

Comment:
Each user is allocated a workspace by default when they sign-up to Konflux for the first time. The cluster on which the workspace is allocated is decided upon when the sign-up is confirmed manually or when automated confirmation is set up. 

Team workspaces are created by creating shared "team accounts", and on-boarding them to Konflux.